rust/lk-sys/rules.mk

```diff
@@ -0,0 +1,43 @@
+# $(error lib/rust_support/rules.mk invoked)
+
+LOCAL_DIR := $(GET_LOCAL_DIR)
+CRATE_NAME := rust_support
+
+MODULE := $(LOCAL_DIR)
+
+# Rust specific module support.
+# Although this starts here, this should be generalized, possibly into a module_rust.mk.
+
+MODULE_SRCDIR := $(MODULE)
+MODULE_BUILDDIR := $(call TOBUILDDIR,$(MODULE_SRCDIR))
+
+# TODO These should be set by the arch.
+RUST_TARGET := aarch64-unknown-none
```

Comment:
TODO: figure out how we can define our own triple and build the core/alloc crates ourselves, as we talked about a bit the other day. The generation of floating point code with the default triple is already technically a major problem as it stands, since there's nothing that keeps LLVM from generating floating point code in the pl011 irq handler with the standard triple.

Reply:
I would also like to address doing this in a more complete way with an additional PR(s).